source/common/network/io_socket_handle_impl.cc

```diff
@@ -499,5 +499,19 @@ Event::FileEventPtr IoSocketHandleImpl::createFileEvent(Event::Dispatcher& dispa
 Api::SysCallIntResult IoSocketHandleImpl::shutdown(int how) {
   return Api::OsSysCallsSingleton::get().shutdown(fd_, how);
 }
+
+absl::optional<std::chrono::milliseconds> IoSocketHandleImpl::lastRoundTripTime() {
+#ifdef TCP_INFO
+  struct tcp_info ti;
+  socklen_t len = sizeof(ti);
+  if (!SOCKET_FAILURE(
+          Api::OsSysCallsSingleton::get().getsockopt(fd_, IPPROTO_TCP, TCP_INFO, &ti, &len).rc_)) {
```

Comment:
@eziskind you might want to chat to @fcfort about generalizing this for https://github.com/envoyproxy/envoy/issues/13166

Reply:
For visibility I have pushed this PR https://github.com/envoyproxy/envoy/pull/13430